src/DoctrineObject.php

```diff
@@ -558,10 +558,12 @@ protected function handleTypeConversions($value, $typeOfField)
                 break;
             case 'string':
             case 'text':
-            case 'bigint':
-            case 'decimal':
                 $value = (string) $value;
                 break;
+            case 'decimal':
+                $value = (float) $value;
+                break;
+            case 'bigint':
```

Comment:
bigints must always be cast to string, because their value range exceeds PHPs range that can represent integers.